File: Reqnroll/Chapter13/13.1-TheLocalScenarioContextPattern/Before/WIMP.LocalScenarioContextSample.App/Infrastructure/UserSession.cs
namespace WIMP.LocalScenarioContextSample.App.Infrastructure;

/// <summary>
/// As simple simulation of a user session, using AsyncLocal to store the current session
/// for the current async context. In a real application, this would likely be more complex
/// and involve actual user authentication and session management.
/// </summary>
public class UserSession
{
    private static readonly AsyncLocal<UserSession> current = new();

    public static UserSession Current
    {
        get
        {
            current.Value ??= new UserSession();
            return current.Value;
        }
    }

    public string? LoggedInCustomerName { get; set; }
}

File: CucumberJVM/Chapter13/13.1-TheLocalScenarioContextPattern/After/src/main/java/com/wimp/localscenariocontext/app/infrastructure/UserSession.java
package com.wimp.localscenariocontext.app.infrastructure;

/**
 * A simple simulation of a user session, using ThreadLocal to store the current session
 * for the current thread. In a real application, this would likely be more complex
 * and involve actual user authentication and session management.
 */
public class UserSession {
    private static final ThreadLocal<UserSession> current = ThreadLocal.withInitial(UserSession::new);

    public static UserSession getCurrent() {
        return current.get();
    }

    private String loggedInCustomerName;

    public String getLoggedInCustomerName() {
        return loggedInCustomerName;
    }

    public void setLoggedInCustomerName(String loggedInCustomerName) {
        this.loggedInCustomerName = loggedInCustomerName;
    }
}
